integration-testing.md

```diff
@@ -0,0 +1,606 @@
+  application.
+- `testsuite/` – The test suite.
+- `tmp/` – A writable directory for storing temporary data during test execution.
+- `sandbox-<hash>/` – The folder where each Magento instance stores temporary and configuration data.
+- `phpunit.xml.dist` – A PHPUnit configuration file.
+
+<a href="#using-data-providers-in-integration-tests"></a>
+## Using Data Providers in Integration Tests
+
+Data providers are a powerful feature of PHPUnit that allow you to run a test multiple times with different data sets.
+They are methods that return an array of arrays, where each array is the set of parameters that will be used for a run
+of the test.
+
+<a href="#creating-a-data-provider"></a>
+### Creating a Data Provider
+
+A data provider method must be public and either return an array of arrays or an object that implements the `Iterator`
+interface and yields an array for each iteration. Each array is a set of parameters for the test method.
+
+Here's an example of a data provider:
+
+```php
+public function provider()
+{
+    return [
+        ['value1', 'value2'],
+        ['value3', 'value4'],
+    ];
+}
+```
+
+<a href="#using-a-data-provider"></a>
+### Using a Data Provider
+
+To use a data provider, add the `@dataProvider` annotation to your test method and specify the name of the data provider
+method. The test method will then be run once for each set of parameters returned by the data provider.
+
+```php
+/**
+ * @dataProvider provider
+ */
+public function testMethod($param1, $param2)
+{
+    // Test code here
+}
+```
+
+In this example, `testMethod` would be run twice: once with `$param1` set to `'value1'` and `$param2` set to `'value2'`,
+and once with `$param1` set to `'value3'` and `$param2` set to `'value4'`.
+
+<a href="#benefits-of-using-data-providers"></a>
+### Benefits of Using Data Providers
+
+Data providers can help you write more efficient and effective tests. They allow you to easily test a method with a
+variety of data, which can help you ensure that your code works correctly in different scenarios. They also make your
+tests more readable and maintainable, as you can easily see the data that is being used for each test and add or remove
+data sets as needed.
+
+<a href="#using-fixtures-in-integration-tests"></a>
+## Using Fixtures in Integration Tests
+
+In the context of testing, a fixture is a fixed state of a set of objects used as a baseline for running tests. The
+purpose of a test fixture is to ensure that there is a well-known and fixed environment in which tests are run so that
+results are repeatable. Examples of fixtures:
+
+- Preparation of input data and setup/creation of fake or mock objects
+- Loading a database with a specific, known set of data
+- Copying a specific known set of files creating a test fixture will create a set of objects initialized to certain
+  states.
+
+<a href="#creating-fixtures"></a>
```

Comment:
I think if there is a section on how to create a fixture, there should also be a path where to place it. But I have no clue where that should be. I guess in the `dev/tests/integration/testsuite` folder somewhere? I have never done this personally.

Reply:
It couldn't be uglier. You have to add your whole relative path to it. 

```php
/**
 * @magentoDataFixture ../../../../app/code/Vendor/Module/Test/Integration/_files/my_custom_fixture.php
 */
public function testMethod()
{
    // Test code here
}

```